src/agent/onefuzz-task/src/tasks/coverage/dotnet.rs

```diff
@@ -0,0 +1,421 @@
+            .unwrap_or(DEFAULT_TARGET_TIMEOUT)
+    }
+}
+
+pub struct DotnetCoverageTask {
+    config: Config,
+    poller: InputPoller<Message>,
+}
+
+impl DotnetCoverageTask {
+    pub fn new(config: Config) -> Self {
+        let poller = InputPoller::new("dotnet_coverage");
+        Self { config, poller }
+    }
+
+    pub async fn run(&mut self) -> Result<()> {
+        info!("starting dotnet_coverage task");
+        self.config.coverage.init_pull().await?;
+
+        let dotnet_path = dotnet_path()?;
+        let dotnet_coverage_path = dotnet_coverage_path()?;
+
+        let heartbeat = self.config.common.init_heartbeat(None).await?;
+        let mut context = TaskContext::new(
+            &self.config,
+            heartbeat,
+            dotnet_path,
+            dotnet_coverage_path.clone(),
+        );
+
+        if !context.uses_input() {
+            bail!("input is not specified on the command line or arguments for the target");
+        }
+
+        context.heartbeat.alive();
+
+        let coverage_local_path = self.config.coverage.local_path.canonicalize()?;
+        let intermediate_files_path = intermediate_coverage_files_path(&coverage_local_path)?;
+        fs::create_dir_all(&intermediate_files_path).await?;
+        let timeout = self.config.timeout();
+        let coverage_dir = self.config.coverage.clone();
+        let dotnet_coverage_path = dotnet_coverage_path;
+
+        tokio::spawn(async move {
+            if let Err(e) = start_directory_monitor(
+                &intermediate_files_path,
+                &coverage_local_path,
+                timeout,
+                coverage_dir,
+                &dotnet_coverage_path,
+            )
+            .await
+            {
+                error!("Directory monitor failed: {}", e);
+            }
+        });
+
+        for dir in &self.config.readonly_inputs {
+            debug!("recording coverage for {}", dir.local_path.display());
+
+            dir.init_pull().await?;
+            let dir_count = context.record_corpus(&dir.local_path).await?;
+
+            info!(
+                "recorded coverage for {} inputs from {}",
+                dir_count,
+                dir.local_path.display()
+            );
+
+            context.heartbeat.alive();
+        }
+
+        context.heartbeat.alive();
+
+        if let Some(queue) = &self.config.input_queue {
+            info!("polling queue for new coverage inputs");
+
+            let callback = CallbackImpl::new(queue.clone(), context)?;
+            self.poller.run(callback).await?;
+        }
+
+        Ok(())
+    }
+}
+
+async fn start_directory_monitor(
+    intermediate_files_path: &PathBuf,
+    coverage_local_path: &Path,
+    timeout: Duration,
+    coverage_dir: SyncedDir,
+    dotnet_coverage_path: &Path,
+) -> Result<()> {
+    info!(
+        "Starting dotnet coverage intermediate file directory monitor on {}",
+        intermediate_files_path.to_string_lossy()
```

Comment:
ICYDK: `Path::display()` exists for interpolating a `Path`/`PathBuf` into `{}`. It doesn't really matter what you use (I think it all boils down to what you have here), but can be a nice semantic hint, depend on the context.